xdsl/dialects/llvm.py

```diff
@@ -46,6 +48,141 @@ def parse_parameters(parser: BaseParser) -> list[Attribute]:
         return [StringAttr.from_str(""), ArrayAttr.from_list(params)]
 
 
+@irdl_attr_definition
+class LLVMPointerType(ParametrizedAttribute, MLIRType):
+    name = "llvm.ptr"
+
+    type: ParameterDef[Attribute | NoneAttr]
+    addr_space: ParameterDef[IntAttr | NoneAttr]
+
+    def print_parameters(self, printer: Printer) -> None:
+        if isinstance(self.type, NoneAttr):
+            return
+
+        printer.print_string("<")
+        printer.print_attribute(self.type)
+        if not isinstance(self.addr_space, NoneAttr):
+            printer.print_string(", ")
+            printer.print_attribute(self.addr_space)
+
+        printer.print_string(">")
+
+    @staticmethod
+    def parse_parameters(parser: BaseParser) -> list[Attribute]:
+        if not parser.tokenizer.starts_with('<'):
+            return [NoneAttr(), NoneAttr()]
+        parser.parse_characters('<', "llvm.ptr parameters expected")
+        type = parser.try_parse_type()
+        if type is None:
+            parser.raise_error(
+                "Expected first parameter of llvm.ptr to be a type!")
+        if not parser.tokenizer.starts_with(','):
+            parser.parse_characters('>',
+                                    "End of llvm.ptr parameters expected!")
+            return [type, NoneAttr()]
+        parser.parse_characters(',', "llvm.ptr args must be separated by `,`")
+        addr_space = parser.parse_int_literal()
+        parser.parse_characters('>', "End of llvm.ptr parameters expected!")
+        return [type, IntegerAttr.from_params(addr_space, IndexType())]
+
+    @staticmethod
+    def untyped():
```

Comment:
Isn't this usually called `opaque` in the LLVM lingo?

Reply:
I think it's not exactly untyped, but rather the type is not introspectable (hence, opaque).

Reply:
Good point! I like that even more!